sdk/storage/azure-storage-file/samples/file_samples_hello_world_async.py

```diff
@@ -0,0 +1,92 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: file_samples_hello_world_async.py
+
+DESCRIPTION:
+    These samples demonstrate common scenarios like instantiating a client,
+    creating a file share, and uploading a file to a share.
+
+USAGE:
+    python file_samples_hello_world_async.py
+    Set the environment variables with your own values before running the sample.
+"""
+
+import os
+import asyncio
+
+SOURCE_FILE = './SampleSource.txt'
+DEST_FILE = './SampleDestination.txt'
+
+
+class HelloWorldSamplesAsync(object):
+
+    connection_string = os.getenv('CONNECTION_STRING')
+
+    async def create_client_with_connection_string_async(self):
+        # Instantiate the FileServiceClient from a connection string
+        from azure.storage.file.aio import FileServiceClient
+        file_service = FileServiceClient.from_connection_string(self.connection_string)
+
+    async def create_file_share_async(self):
+        # Instantiate the ShareClient from a connection string
+        from azure.storage.file.aio import ShareClient
+        share = ShareClient.from_connection_string(self.connection_string, share_name="helloworld1")
+
+        # Create the share
+        async with share:
+            await share.create_share()
+
+            try:
+                # [START get_share_properties]
+                properties = await share.get_share_properties()
+                # [END get_share_properties]
+
+            finally:
+                # Delete the share
+                await share.delete_share()
+
+    async def upload_a_file_to_share_async(self):
+        # Instantiate the ShareClient from a connection string
+        from azure.storage.file.aio import ShareClient
+        share = ShareClient.from_connection_string(self.connection_string, share_name='helloworld2')
+
+        # Create the share
+        async with share:
+            await share.create_share()
+
+            try:
+                # Instantiate the FileClient from a connection string
+                # [START create_file_client]
+                from azure.storage.file.aio import FileClient
+                file = FileClient.from_connection_string(
+                    self.connection_string,
+                    share_name='helloworld2',
+                    file_path="myfile")
+                # [END create_file_client]
+
+                # Upload a file
+                async with file:
+                    with open(SOURCE_FILE, "rb") as source_file:
+                        await file.upload_file(source_file)
```

Comment:
Double checking that we don't need start/end here?  (Still reconciling myself with our docs generation, I THINK you should be fine as long as the function itself becomes a sample, but with all the individual start/end snippets upstream I'm doing a paranoid consistency check that this didn't need to be bracketed too)